Automattic/wp-calypso
# Display coming soon page template if it exists

This will add a coming soon category to the bottom of the starter page templates available on the new page screen. 
Part of #45689

<img width="809" alt="Screen Shot 2020-09-24 at 1 18 58 PM" src="https://user-images.githubusercontent.com/22446385/94105825-78d7d800-fe7d-11ea-8f7d-9754a28e3ddb.png">


### Testing instructions.
Note: currently this card is relatively difficult to test locally because when testing in your sandbox:
> The list of all templates is loaded from php code in [class-starter-page-templates.php#213](https://github.com/Automattic/wp-calypso/blob/master/apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/class-starter-page-templates.php#L213)
> Because this code runs on the server, your hosts file will not be used, it will load the list of all templates from the production public-api.wordpress.com.
> Therefore the new coming soon page will not be included in the results.

However it is possible to test locally like so:
* Checkout this branch
* Set up wp-env following the instructions here: PCYsg-ly5-p2 
* Sandbox `public-api.wordpress.com`
* In another terminal tab, sync the api portion of this code change to your sandbox with `yarn dev --sync`
* On `http://localhost:4013` go to add a new page.
* Scroll to the bottom of the new page template list and a "coming soon" template should be available
* Locally for me, actually creating a page with any template fails, I'm investigating this but still get the same error on master branch so can say it's unrelated to this PR
